source/MRMesh/MRUniteManyMeshes.cpp

```diff
@@ -94,6 +96,10 @@ tl::expected<Mesh, std::string> uniteManyMeshes(
 
     // find non intersecting groups for simple merge instead of union
     std::vector<std::vector<int>> nonIntersectingGroups;
+    std::vector<Box3d> meshesBoxes( meshes.size() );
+    for ( int m = 0; m < meshes.size(); ++m )
+        if ( meshes[m] )
+            meshesBoxes[m] = Box3d( meshes[m]->computeBoundingBox() );
```

Comment:
probably take the box from aabb is better here than recompute it? Mesh.getBoundingBox()

Reply:
i am not sure that aabb tree is already build, and it will be invalid after merging, so calculating aabb tree might be harder than calculating box